### DISCUSSION TITLE
[FEATURE] Parameters for plate editor operations.

### DISCUSSION THREAD

**Discussion by verwers** [idea-statica/ideastatica-public#152] [Features and Ideas]
Would be great if I can edit parameters of a notch.

![image](https://user-images.githubusercontent.com/19746347/138816416-c1de6407-fcc5-47f6-a2c4-efe1ea72d09c.png)

**Comment by vojtech-chalupa** [comment]
Hello Steven, can you please elaborate on your use case? Are you using Parameters to manipulate the geometry of an existing connection? I will get back to you. Still, I am interested in what exactly you are trying to accomplish so that I can understand this request in a broader context. Thanks!

**Comment by verwers** [comment]
The need for this parameterization is that we have a list of shear connections that have connection plates with notches like the example which have to be made in IDEA StatiCa for verification. However, I cannot change the notches, so I run into trouble when changing the beam crosssections.

Later, I want to use this particular workflow to generate a suitable shear connection for a given load case (or cases) using constrained global optimization strategies. This would then mean that for standard connections the entire design process is done by the computer instead of trial-and-error methods by the engineers.

**Comment by vojtech-chalupa** [comment]
Understood, thank you. At the moment, it is not possible using parameters in the plate editor. We'll look into enabling this.
In the meantime, the only workaround I'm coming up with is the following:

I've modeled a simple example (version 21.1) of how you can use negative volumes to create notches. Their position can be modified through parameters, cross-section can also be reassigned.

[Finplate.zip](https://github.com/idea-statica/ideastatica-public/files/7441719/Finplate.zip)

**Comment by verwers** [comment]
Is it possible to use a dxf to generate a cross section for the negative volumes? The square cutout will likely cause high stress concentrations at the corners of the cutouts. I am unable to generate a general crosssection with rounded corners.

**Comment by verwers** [comment]
BETTER WORKAROUND:
use rounded square section to cut. Radius of notch = inner radius + thickness

OLD SOLUTION:
Ok, the workaround is to cut by a combined crosssection which is square with small square notches at the corners. after that, the radius in the plate can be made by a circular cut using a circular crosssection.

**Comment by TareqAli-Graitec** [comment]
@vojtech-chalupa Is this still not possible? I had a look through the [reference guide documentation](https://github.com/idea-statica/ideastatica-public/wiki/Reference-Guide-Expression-Parameters) but still could not find the expressions to use to access the properties and parameterize the plate editing functions.

**Comment by nathanlukeIDEA** [comment]
Hi @TareqAli-Graitec, no unfortunately, this is still not avaliable. I have added feature you can track. https://github.com/idea-statica/ideastatica-public/issues/833
